File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.cpp
class Solution {
public:
    string reverseVowels(string s) {
        vector<char> vowel;
        vector<int> idss;

        for (int i = 0; i < s.size(); i++) {
            if (s[i] == 'a' || s[i] == 'e' || s[i] == 'i' || s[i] == 'o' ||
                s[i] == 'u' || s[i] == 'A' || s[i] == 'E' || s[i] == 'I' ||
                s[i] == 'O' || s[i] == 'U') {
                vowel.push_back(s[i]);
                idss.push_back(i);
            }
        }

        reverse(vowel.begin(), vowel.end());

        for (int i = 0; i < vowel.size(); i++) {
            s[idss[i]] = vowel[i];
        }
        return s;
    }
};

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {

        ArrayList<Character> vowel = new ArrayList<>();
        ArrayList<Integer> ids = new ArrayList<>();

        for(int i=0;i<s.length();i++){

            char ch = Character.toLowerCase(s.charAt(i));

            if(ch=='a' || ch=='e' || ch=='i' || ch=='o' || ch=='u'){
                vowel.add(s.charAt(i));
                ids.add(i);
            }
        }

        Collections.reverse(vowel);

        char[] arr = s.toCharArray();

        for(int i=0;i<vowel.size();i++){
            arr[ids.get(i)] = vowel.get(i);
        }

        return new String(arr);
    }
}